TCAT/Model/Route.swift

```diff
@@ -62,12 +65,26 @@ class Route: NSObject {
         return Time.dateComponents(from: now, to: departureTime)
     }
     
-    var directions: [Direction]
-    var mainStops: [String]
+    var directions: [Direction] = [Direction]()
+    var mainStops: [String] = [String]()
     //N2SELF: ADD TO EXAMPLE
-    var mainStopsNums: [Int] //-1 for pins
-    var travelDistance: Double //of first stop
-    var lastStopTime: Date // the critical last time a bus route runs
+    var mainStopsNums: [Int] = [Int]()//-1 for pins
+    var travelDistance: Double = 0.0 //of first stop
+    var lastStopTime: Date = Date()// the critical last time a bus route runs
+    
+   
+    
+    required init(json: JSON) throws {
+        super.init()
+        print(json)
+        departureTime = timeToDate(time: json["departureTime"].stringValue)
+        arrivalTime = timeToDate(time: json["arrivalTime"].stringValue)
+        directions = directionJSON(json:json["directions"].array!)
+        mainStops = json["stopNames"].arrayObject as! [String]
+        mainStopsNums = json["stopNumbers"].arrayObject as! [Int]
+        travelDistance = (directions[0] as! WalkDirection).travelDistance
```

Comment:
I think it was supposed to be the "25 minutes" for the entire route in the summary view, but I actually just calculate it from arrival and departure time. We can probably get rid of the variable as a whole